src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,152 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> newNode = new Node<>(null, value, null);
+        if (size == 0) {
+            head = newNode;
+        } else {
+            newNode = new Node<>(tail, value, null);
+            tail.next = newNode;
+        }
+        tail = newNode;
+        size++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        indexValidationForAdd(index);
+        if (index == size) {
+            add(value);
+            return;
+        } else {
+            Node<T> nodeAtIndex = findNode(index);
+            Node<T> oldHead = head;
+            if (nodeAtIndex.previous == null) {
+                head = new Node<>(null, value, oldHead);
+            } else {
+                Node<T> newNode = new Node<>(nodeAtIndex.previous, value, nodeAtIndex);
+                newNode.previous.next = newNode;
+                nodeAtIndex.previous = newNode;
+            }
+        }
+        size++;
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T value : list) {
+            add(value);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        return findNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        indexValidation(index);
+        Node<T> nodeToChange = findNode(index);
+        T oldItem = nodeToChange.item;
+        nodeToChange.item = value;
+        return oldItem;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidation(index);
+        Node<T> nodeToRemove = findNode(index);
+        if (size == 1) {
+            head = null;
+            tail = null;
+        } else {
```

Comment:
add return before in order to avoid complex if-else statements

Reply:
fixed